modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/BinaryMetadataFileStore.java

```diff
@@ -117,6 +143,10 @@ void writeMetadata(BinaryMetadata binMeta) {
 
             U.error(log, msg);
 
+            for (GridFutureAdapter fut : writeOpFutures.values()) {
```

Comment:
How is it in sync with writeMetadataAsync? Can we add new future while iterating?
Also, shouldn't we clear this map after? Does it matter?

Reply:
Indeed here was a data race, I closed it with adding additional flag similar to implementation in BinaryMetadataTransport.